etcdmain/config.go

```diff
@@ -127,9 +133,66 @@ type config struct {
 
 	enablePprof bool
 
+	configFile string
+
 	ignored []string
 }
 
+type yamlSecConfig struct {
+	CAFile        string `json:"ca-file"`
+	CertFile      string `json:"cert-file"`
+	KeyFile       string `json:"key-file"`
+	CertAuth      bool   `json:"client-cert-auth"`
+	TrustedCAFile string `json:"trusted-ca-file"`
+}
+
+type yamlConfig struct {
```

Comment:
I feel we should just combine this with config. We expose everything in the pervious config except the one has internal types like clusterState. We keep these with internal types and the one without types (which is public) and do the conversion after parsing if it is read in from configuration file. In that way, we do not have to maintain two set of almost identical configuration structure.